src/linker/Linker/LinkContext.cs

```diff
@@ -484,9 +488,22 @@ public bool IsOptimizationEnabled (CodeOptimizations optimization, AssemblyDefin
 
 		public void LogMessage (MessageContainer message)
 		{
-			if (!LogMessages || message == MessageContainer.Empty)
+			if (message == MessageContainer.Empty)
 				return;
 
+			if (!LogMessages && message.Category == MessageCategory.Warning) {
+				switch (DontWarn) {
+				case NoWarn.All:
+					return;
+
+				case NoWarn.Analysis:
+					if (MessageSubCategory.Analysis.Contains (message.SubCategory))
```

Comment:
Nit: Maybe use `if (Array.IndexOf (MessageSubCategory.Analysis, message.SubCategory) > -1)` to not have to add Linq to the LinkContext? Also, IndexOf does a faster lookup but not much of a difference. I do get that this is more readable, so I'm also ok if you don't want to change it